src/FwdState.cc

```diff
@@ -259,7 +262,8 @@ FwdState::completed()
 
     flags.forward_completed = true;
 
-    request->hier.stopPeerClock(false);
+    if (!flags.tunneled)
+        request->hier.stopPeerClock(false);
```

Comment:
I am concerned that FwdState code implies that completed() may, in some cases, be called from FwdState destructor. FwdState is reference-counted. Thus, completed() method may not be called for a while if some long-lived object is storing a reference-counting pointer to FwdState...

Can we stopPeerClock() in "ends forwarding" method called stopAndDestroy() instead?